### PR TITLE
Switch CI from bors to merge queue

### PR DESCRIPTION
**Pull request by Kobzol**
Context: https://rust-lang.zulipchat.com/#narrow/channel/242791-t-infra/topic/Reducing.20the.20amount.20of.20repos.20that.20use.20bors.2Fhomu

This should be merged with cooperation from an infra-admin.

Opening as a draft, because the CI on this repo is.. interesting :) So I want to make sure that the changes make sense. I renamed `clippy_bors.yml` to `clippy.yml` and `clippy.yml` to `clippy_pr.yml`, which makes the diff a bit.. tricky.

changelog: Move Clippy from bors to merge queues

### CODE REVIEW

File: .github/workflows/clippy.yml
Comment (flip1995) [inline .github/workflows/clippy.yml:1]: Could you name this file `clippy_mq.yml`. With that it should also be way easier to review.

File: README.md
Comment (Kobzol) [reply README.md:3]: No, we don't want to run anything on master. Actually, with a merge queue, I think that the badge is redundant, lol. By definition, master has to be green, otherwise there would be a bug in the merge queue.. :)

File: .github/workflows/clippy_pr.yml
Comment (Kobzol) [reply .github/workflows/clippy_pr.yml:4]: I see. That's a bit problematic with the merge queue, because it creates separate branches for each merge queue test. But we don't want to run CI twice on these. I'm not sure if there's a single prefix that we could use here to ignore these.

We could potentially just use a `push:` hook and then somehow skip the workflow if the repository is rust-lang/rust-clippy :thinking:

File: README.md
Comment (flip1995) [reply README.md:3]: > (The same was true for bors, btw)

Yeah, I know. I just like green badges :D

Then I'd say, let's just remove the badge, as I don't think we can generate any badge that makes sense with merge queues?

File: .github/workflows/clippy_pr.yml
Comment (flip1995) [reply .github/workflows/clippy_pr.yml:4]: > We could potentially just use a `push:` hook and then somehow skip the workflow if the repository is rust-lang/rust-clippy

That sounds a bit too hacky. TBH I never looked at CI results in my fork. Not sure anymore who requested that. We could do the scream test and just remove it and see if anyone complains :smiling_imp:

Comment (Kobzol) [reply .github/workflows/clippy_pr.yml:4]: I agree. Usually the only interaction of CI in forks that I see is people being annoyed by an e-mail flurry spam about failing CI workflows after forking something.

File: README.md
Comment (Kobzol) [reply README.md:3]: I suppose that we can simply avoid the `?branch` and `event` parametrs, and just use the name of the CI job that runs on merge queues, that should work.

File: .github/workflows/clippy_pr.yml
Comment (y21) [reply .github/workflows/clippy_pr.yml:4]: I have a PR on my fork where I do all kinds of experiments so I can see the lintcheck results (the lintcheck workflow only runs on PRs), and I believe `@xFrednet` mentioned on Zulip they do the same IIRC. But seeing how the lintcheck.yml file is unchanged, I assume this is unaffected and will still work?

Comment (Kobzol) [reply .github/workflows/clippy_pr.yml:4]: Indeed, lintcheck is untouched by this PR.

File: README.md
Comment (Kobzol) [reply README.md]: Pushed a fix to the badge URLs.

**Comment by Kobzol** [comment]
Not sure if I understand the CI error: https://github.com/rust-lang/rust-clippy/actions/runs/11513171989/job/32049393605#step:6:39. Does it actually check that the link returns 200? In that case, we'll need to update the badge only after merging this.

**Comment by flip1995** [comment]
Yes, it checks that all links in our markdown files actually lead to something (this check is actually copied from the Rust repo, so that we don't run into problems during sync).

File: .github/workflows/clippy_pr.yml
Comment (flip1995) [reply .github/workflows/clippy_pr.yml:4]: Also: If you open a PR in a fork (instead of just pushing a branch) all checks will run anyway. This is more about not having to open a PR in your fork to get the clippy workflow result.

**Comment by Kobzol** [comment]
Ok, I removed the badge URL changes, we can fix them up after this is merged.

**Comment by Kobzol** [comment]
CI failure looks spurious.

**Comment by flip1995** [comment]
I restarted the run. It was spurious.

**Review by flip1995 — APPROVED** [review]

**Review by flip1995 — APPROVED** [review]
LGTM. We decided to remove the path filters for PRs and run every workflow on every PR: clippy test and clippy dev test ran anyway on almost all PRs and Remark is really quick to run. This way it is possible to require PR CI to be green, before the PR is put into the merge queue. The reviewer still doesn't have to wait for the PR to be green, but can press "merge when ready" before that.

**Pull request merged**
